platform/linux-generic/odp_pool.c

```diff
@@ -1673,6 +1683,58 @@ int odp_pool_stats(odp_pool_t pool_hdl, odp_pool_stats_t *stats)
 	return 0;
 }
 
+int odp_pool_stats_selected(odp_pool_t pool_hdl, odp_pool_stats_selected_t *stats,
+			    odp_pool_stats_opt_t *opt)
+{
+	pool_t *pool;
+
+	if (odp_unlikely(pool_hdl == ODP_POOL_INVALID)) {
+		_ODP_ERR("Invalid pool handle\n");
+		return -1;
+	}
+	if (odp_unlikely(stats == NULL)) {
+		_ODP_ERR("Output buffer NULL\n");
+		return -1;
+	}
+	if (odp_unlikely(opt == NULL)) {
+		_ODP_ERR("Pool counters NULL\n");
+		return -1;
+	}
+
+	pool = _odp_pool_entry(pool_hdl);
+
+	if (opt->all & ~pool->params.stats.all) {
+		_ODP_ERR("Trying to read non-enabled counter\n");
+		return -1;
+	}
+
+	if (opt->bit.available)
+		stats->available = ring_ptr_len(&pool->ring->hdr);
+
+	if (opt->bit.alloc_ops)
+		stats->alloc_ops = odp_atomic_load_u64(&pool->stats.alloc_ops);
+
+	if (opt->bit.alloc_fails)
+		stats->alloc_fails = odp_atomic_load_u64(&pool->stats.alloc_fails);
+
+	if (opt->bit.free_ops)
+		stats->free_ops = odp_atomic_load_u64(&pool->stats.free_ops);
+
+	if (opt->bit.total_ops)
+		stats->total_ops = stats->alloc_ops + stats->free_ops;
```

Comment:
alloc_ops / free_ops are unintialized if application didn't request those counters, but only this one.

Reply:
Fixed in V3.